app/controllers/pupils/public_displays_controller.rb

```diff
@@ -0,0 +1,146 @@
+module Pupils
+  class PublicDisplaysController < ApplicationController
+    load_resource :school
+
+    include SchoolAggregation
+    include ActionView::Helpers::TagHelper
+    include ActionView::Context
+
+    skip_before_action :authenticate_user!
+    before_action :check_aggregated_school_in_cache, except: :index
+    before_action :set_fuel_type, except: :index
+    before_action :check_fuel_type, only: :charts
+    before_action :set_analysis_dates, except: :index
+
+    layout 'public_displays'
+
+    rescue_from StandardError do |exception|
+      Rollbar.error(exception, school: @school.name, school_id: @school.id)
+      raise unless Rails.env.production?
+      locale = LocaleFinder.new(params, request).locale
+      I18n.with_locale(locale) do
+        render 'error', status: :bad_request
+      end
+    end
+
+    def index
+      render 'index', layout: 'application'
+    end
+
+    def equivalences
+      meter_types = case @fuel_type
+                    when :electricity
+                      [:electricity, :solar_pv]
+                    else
+                      [@fuel_type]
+                    end
+      @equivalence = equivalence_for_meter_types(meter_types)
+    end
+
+    def charts
+      raise 'Non-public data' unless @school.data_sharing_public?
+      raise 'Not data enabled' unless @school.data_enabled?
+      @chart_type = params.require(:chart_type).to_sym
+      @chart = find_chart(@fuel_type, @chart_type)
+    end
+
+    private
+
+    def set_fuel_type
+      @fuel_type = params.require(:fuel_type).to_sym
+    end
+
+    def check_fuel_type
```

Comment:
I think this is something that could be a method in the school model, at some point anyway.